### PR TITLE
sphinx_rtd_theme/layout: Set url_root properly on index, don't use '#'

### PR DESCRIPTION
**Pull request by rkdarst**
- You can see a practical demonstration of the problem, fully seeing
  the effect if it is not fixed, here (though that example uses the
  alabaster theme, the effect is the same here):
  https://github.com/executablebooks/sphinx-copybutton/issues/110
- This is a copy of a fix from Sphinx.  The sphinx pull request is
  https://github.com/sphinx-doc/sphinx/pull/8524

Detailed description:

- url_root is set to `#` on the index page, which layout.html tries to
  change back to `''` (the empty string).
- But, this updated url_root wasn't used in the actual location, as an
  argument to `documentation_options.js`.
- Thus, clever enough templates, which tried to use
  `$DOCUMENTATION_OPTIONS.URL_ROOT` inside javascript would fail.
  This was manifested as broken links, which led to this issue:
  https://github.com/executablebooks/sphinx-copybutton/issues/110
- I have eventually traced that back to sphinx itself, and found that
  layout.html tried to fix the problem, but the fixed value wasn't
  used.
- This fix works in my basic test, but I will continue with more tests.
- Review:
  - someone more clever should examine this and make sure it makes
    sense
  - This does not have tests.  Should it?

### CODE REVIEW

**Comment by stsewd** [comment]
Thanks, I'm not that familiar with all this, but makes sense. I'll merge after is approved upstream https://github.com/sphinx-doc/sphinx/pull/8524

**Comment by rkdarst** [comment]
Merged upstream now.

**Review by stsewd — APPROVED** [review]

**Comment by rkdarst** [comment]
thanks!

**Pull request merged**
